src/xdai-bots/x-kleros-liquid.js

```diff
@@ -0,0 +1,178 @@
+const delay = require('delay')
+const _xKlerosLiquid = require('../contracts/x-kleros-liquid.json')
+
+const DELAYED_STAKES_ITERATIONS = 15
+
+module.exports = async (web3, batchedSend) => {
+  // Instantiate the Kleros Liquid contract.
+  const klerosLiquid = new web3.eth.Contract(
+    _xKlerosLiquid.abi,
+    process.env.XDAI_X_KLEROS_LIQUID_CONTRACT_ADDRESS
+  )
+  const PhaseEnum = Object.freeze({ staking: 0, generating: 1, drawing: 2 })
+
+  // Keep track of executed disputes so we don't waste resources on them.
+  const executedDisputeIDs = {}
+
+  // Run every 5 minutes.
+  while (true) {
+    // Try to execute delayed set stakes if there are any. We check because this transaction still succeeds when there are not any and we don't want to waste gas in those cases.
+    if (
+      (await klerosLiquid.methods.lastDelayedSetStake().call()) >=
+      (await klerosLiquid.methods.nextDelayedSetStake().call())
+    )
+      batchedSend({
+        args: [DELAYED_STAKES_ITERATIONS],
+        method: klerosLiquid.methods.executeDelayedSetStakes,
+        to: klerosLiquid.options.address
+      })
+
+    // Loop over all disputes.
+    try {
+      let disputeID = 0
+      while (true) {
+        if (!executedDisputeIDs[disputeID]) {
+          const dispute = await klerosLiquid.methods.disputes(disputeID).call()
```

Comment:
disputes is a mapping instead of an array now, which means that fetching an out of bounds item won't throw but return an empty dispute. I see two options here: (1) if dispute.arbitrated is 0x0, manually throw or (2) query [totalDisputes](https://github.com/kleros/xdai-kleros-liquid/blob/master/contracts/kleros/xKlerosLiquid.sol#L180) before each loop and don’t rely on the code throwing an error to find the last dispute.

Reply:
I see, thanks for pointing that out... I overlooked that part.